Source/LibationFileManager/Templates/SeriesOrder.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+#nullable enable
+namespace LibationFileManager.Templates;
+
+public class SeriesOrder : IFormattable
+{
+	public object[] OrderParts { get; }
+	private SeriesOrder(object[] orderParts)
+	{
+		OrderParts = orderParts;
+	}
+
+	public override string ToString() => ToString(null, null);
+
+	/// <summary>
+	/// Use float formatters to format the number parts of the order.
+	/// </summary>
+	public string ToString(string? format, IFormatProvider? formatProvider)
+		=> string.Concat(OrderParts.Select(p => p is float f ? f.ToString(format) : p.ToString())).Trim();
+
+	public static SeriesOrder Parse(string? order)
+	{
+		List<object> parts = new();
+		while (TryParseNumber(order, out var value, out var range))
+		{
+			var prefix = order[..range.Start.Value];
+			if(!string.IsNullOrWhiteSpace(prefix))
+				parts.Add(prefix);
+
+			parts.Add(value);
+
+			order = order[range.End.Value..];
+		}
+
+		if (!string.IsNullOrWhiteSpace(order))
+			parts.Add(order);
+
+		return new(parts.ToArray());
+	}
+
+	/// <summary>
+	/// Try to parse any positive number from within the string (greedy).
+	/// </summary>
+	/// <param name="numString">the string to search for a numeric value</param>
+	/// <param name="value">If this function succeeds, the number that was found; otherwise zero.</param>
+	/// <param name="range">If this function succeeds, the range of characters representing <paramref name="value"/> in <paramref name="numString"/>; otherwise default</param>
+	/// <returns>True if a number was found; otherwise false.</returns>
+	private static bool TryParseNumber([NotNullWhen(true)] string? numString, out float value, out Range range)
+	{
+		value = 0;
+		if (string.IsNullOrWhiteSpace(numString))
+		{
+			range = default;
+			return false;
+		}
+
+		for (int s = 0; s < numString.Length; s++)
+		{
+			//Assume any valid number will begin with a digit.
+			//This way, leading dots and dashes will never be considered part of a number, so
+			//no negative series numbers and no fractional series numbers < 1 (unless preceded with a '0').
+			if (!char.IsDigit(numString[s]))
+				continue;
+
+			for (int e = numString.Length; e > s; e--)
+			{
+				//The float parser will succeed with trailing whitespace,
+				//but we want to preserve it in the final display string.
+				if (char.IsWhiteSpace(numString[e - 1]))
+					continue;
+
+				var substring = numString[s..e];
+				if (float.TryParse(substring, out value))
```

Comment:
@rmcrackan I ran into [an issue](https://github.com/Mbucari/GEHistoricalImagery/issues/6) on one of my other projects earlier this year which was causing floating point numbers to fail parsing due to a Culture mismatch. My program required float numbers to use the dot as a decimal separator because it used a comma as a delimiter, but then parsing failed in cultures that use a comma as a decimal separator.

I don't know if series numbers are actually ever floating point numbers, and if they are, I don't know if they use culture-dependent decimal separators. My _guess_ is that they would use a dot for everyone. If that's the case, then we should specify the invariant culture when parsing.

Reply:
There are tons of books with a dot in their series order. eg 2.1, 2.2, 3.1 here: https://www.audible.com/series/B07JZCTPDR

Reply:
Thanks for the link. Yes, this is going to be a problem.

This book is series order "2.5": https://www.audible.fr/pd/B07F88XZ3V 

I queried the catalog page for that title using the french AudibleApi locale, and it indeed did return "2.5", meaning that the french localization won't parse that as a float.

This isn't a problem for any of the Dtos because the json deserializer uses the invariant culture by default, but this is a problem for my new series parser and potentially other parsing operations throughout Libation.